src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -2,80 +2,261 @@
+        }
         // check that student is not already enrolled into this section
+        Enrollment existing = enrollmentRepository.findEnrollmentBySectionNoAndStudentId(sectionNo, student.getId());
+        if(existing != null){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "this student is already enrolled in this section");
+        }
+
         // create a new enrollment entity and save.  The enrollment grade will
         // be NULL until instructor enters final grades for the course.
+        Enrollment e = new Enrollment();
+        e.setGrade(null);
+        e.setStudent(u);
+        e.setSection(s);
 
-        // remove the following line when done.
-        return null;
+        enrollmentRepository.save(e);
 
+        return new EnrollmentDTO(
+                e.getEnrollmentId(),
+                e.getGrade(),
+                e.getStudent().getId(),
+                e.getStudent().getName(),
+                e.getStudent().getEmail(),
+                e.getSection().getCourse().getCourseId(),
+                e.getSection().getCourse().getTitle(),
+                e.getSection().getSecId(),
+                e.getSection().getSectionNo(),
+                e.getSection().getBuilding(),
+                e.getSection().getRoom(),
+                e.getSection().getTimes(),
+                e.getSection().getCourse().getCredits(),
+                e.getSection().getTerm().getYear(),
+                e.getSection().getTerm().getSemester()
+        );
     }
 
     // student drops a course
     // user must be student
+
+    //this will be the logged in user? Is this for future use after login / pwd is implemented?
+
    @DeleteMapping("/enrollments/{enrollmentId}")
-   public void dropCourse(@PathVariable("enrollmentId") int enrollmentId) {
+   @PreAuthorize("hasAuthority('SCOPE_ROLE_STUDENT')")
+   public void dropCourse(@PathVariable("enrollmentId") int enrollmentId, Principal principal) {
+
+       //retrieve enrollment by Id
+       Enrollment e = enrollmentRepository.findById(enrollmentId).orElse(null);
+       if(e == null){
+           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "invalid enrollment id");
+       }
+
+       //retrieve section by section number in enrollment
+       Section s = sectionRepository.findById(e.getSection().getSectionNo()).orElse(null);
+       if (s == null) {
+           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "sectionNo not a valid section number");
+       }
+
+       //retrieve term in section
+       Term t = termRepository.findById(s.getTerm().getTermId()).orElse(null);
+       if (t == null) {
+           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "invalid term");
+       }
+
+       //retrieve user by getting student ID if user is a student, set to null is not student
+       User u = userRepository.findById(e.getStudent().getId()).orElse(null);
+
```

Comment:
Should we be matching that the user who called the method matches the enrollment?